src/util/AsioHelpers.h

```diff
@@ -167,14 +183,13 @@ inline T runAndWaitForAwaitable(net::awaitable<T> awaitable,
   auto future = net::co_spawn(net::make_strand(ioContext), std::move(awaitable),
                               net::use_future);
 
-  while (true) {
-    auto futureStatus = future.wait_for(std::chrono::milliseconds{0});
-    if (futureStatus == std::future_status::ready) {
-      break;
-    }
+  std::future_status futureStatus;
+  do {
+    ioContext.poll();
```

Comment:
This is an optimization for the only use-case we currently have: Running stuff on an io_context without attached threads where an awaitable only spawns a limited amount of other tasks. If we'd pass an awaitable that posts new task that spawn itself new tasks forever so that the queue is never completely empty, this call to poll will never return.
But for our common case a single call to poll will just run most of the stuff that is currently pending so chances are we'll never wait 5ms for nothing.

The alternative would be to keep using poll_one, and instead check the return value of it and only sleep for 5ms if nothing was actually executed.

Reply:
What you are doing is really a little bit dangerous, as the current thread basically becomes a worker thread for the context and might be stalled forever:). 
So I think the cleanes loop wold be 

```
while (true) {
n = poll_one();
if (future is ready) {break;}
if (n < 1) {sleep;}
}
```
This keeps the behavior as it was, but still makes your unit tests faster:)